### ISSUE TITLE
Rebrand to "onboard"

### ISSUE THREAD

**Issue by andrewda** [SouthEugeneRoboticsTeam/github-invite#1]
This program should do more than just GitHub invites. It should invite to Slack, add to Google Drive and maybe find a way to add to Team Cowboy.

**Comment by andrewda** [comment]
For Slack: https://github.com/ErikKalkoken/slackApiDoc/blob/master/users.admin.invite.md
For Drive: https://developers.google.com/drive/api/v3/reference/permissions/create

Looks like Team Cowboy doesn't have a way to invite users through their API 🙁.

**Comment by andrewda** [comment]
Process:

1. Sign in with `@4j.lane.edu` email address
2. Enter email address you'd like to use for Slack/Google Drive
3. Enter GitHub username (if applicable)

**Comment by will-hou** [comment]
I was playing around with [Zapier](https://zapier.com) to see if it could handle automatic on-boarding really easily for us. It seems to integrate really well with Google Sheets/Forms and can send slack invite and welcome emails whenever a new row is added to a spreadsheet. Do you think this would be good enough for us? 

Creating our own program would be really cool though and definitely add more flexibility.

**Comment by andrewda** [comment]
https://zapier.com/pricing

It costs $20/mo to do anything that isn't directly 1:1. Seems like this will be more like 1:3 as we'll be inviting them to three different programs. Do we really want to spend $20/mo for something we can do in-house for free?

If I'm misunderstanding the pricing let me know, though!

**Comment by andrewda** [comment]
Also I don't see a way on Zapier to add someone to a GitHub organization or to a Shared Google Drive (though it's likely I'm just not looking hard enough)

**Comment by will-hou** [comment]
You're totally right on the pricing. I was thinking of making one seperate 'zap' for each action we want to have happen. It's not the most efficient way to do things but it should solve the cost problem. We can only have a maximum of 5 active 'zaps' on an account I believe but we probably won't have that many actions to perform apart from slack and team drive invites.

**Comment by will-hou** [comment]
I actually didn't find a way to add someone to a shared drive with zapier either but I haven't looked very hard either. I'll look more into it tonight

Also if you or someone else want to work on an automatic on-boarder, please feel free to! It really does sound like a great project. I just wanted y'all to know that there is a potential alternative in case we want to save some time

**Comment by will-hou** [comment]
Sorry for the delay. From what I could find, it seems that zapier doesn't have a built-in way to invite people to a Team Drive. However, it does give us the option of triggering custom code that we write to hit an API. I'll try to get that working and will update when I can!

**Comment by andrewda** [comment]
Okay! Though the Google API is pretty nasty in terms of authentication so
we might have to write our own backend anyways (and at that point may as
well just host the frontend too).

**Comment by will-hou** [comment]
Ugh tell me about it. I'm trying to authenticate using Oauth service accounts but I can't seem to format the JWT signature properly. Zapier doesn't support any external libraries so I'm trying to just make it work using requests. 

Also, they limit free users to only 1s of runtime and 128mb of RAM at a time so we might have to end up playing some code golf haha.

I'll play around with it a bit more but it seems that zapier will probably not be able to handle team drive invites for us (not without a lot more effort anyway)
